elastic/integrations
# AWS Bedrock add runtime dataset

## Enhancement

AWS Bedrock runtime metrics include `Invocations`, `InvocationLatency`,
`InvocationClientErrors`, `InvocationServerErrors`, `OutputTokenCount`,
`OutputImageCount`, `InvocationThrottles`. These metrics can be used for a variety of use cases including 
 - Comparing model latency. 
 - Measuring input and output token counts.
 - Detecting the number of invocations that the system throttled.

The enhancement is to create a new dataset named `runtime` to collect the runtime metrics of AWS Bedrock. 

## Proposed commit message

Enhance the existing AWS Bedrock integration package to capture runtime metrics. 

## Checklist


## Author's Checklist

- Tested with ES stack version 8.13


## How to test this PR locally

- elastic-package build
- elastic-package stack up -v -d --services package-registry
- Configure AWS Bedrock
- Use AWS Playground to run prompts related to text, chat, images
- Verify with dashboard data 
- Verify with the records populated under the Discover view.
## Related issues

- Closes https://github.com/elastic/integrations/issues/10323

## Screenshots
<img width="2557" alt="Screenshot 2024-07-10 at 6 18 39 PM" src="https://github.com/elastic/integrations/assets/101976829/9859b1c1-43f1-4df6-bb36-71a6039b32ee">


![Screenshot 2024-07-10 at 6 19 47 PM](https://github.com/elastic/integrations/assets/101976829/4d85b921-2217-4cd3-8ccc-cbd1dd2a81bb)

![Screenshot 2024-07-10 at 6 20 26 PM](https://github.com/elastic/integrations/assets/101976829/18c75bc9-fb0d-4bd2-971e-15762e94ee8c)
